coreneuron/network/partrans.cpp

```diff
@@ -119,33 +114,33 @@ void nrnthread_v_transfer(NrnThread* _nt) {
     int* insrc_indices = ttd.insrc_indices.data();
     double* tar_data = _nt->_data;
     // last element in the displacement vector gives total length
+#if defined(CORENEURON_ENABLE_GPU) && !defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && \
+    defined(_OPENACC)
     int n_insrc_buf = insrcdspl_[nrnmpi_numprocs];
     int ndata = _nt->_ndata;
+#endif
 
-    // clang-format off
-
-    #pragma acc parallel loop present(  \
-        insrc_indices[0:ntar],          \
-        tar_data[0:ndata],              \
-        insrc_buf_[0:n_insrc_buf])      \
-    if (_nt->compute_gpu)               \
-        async(_nt->stream_id)
-    // clang-format on
+    nrn_pragma_acc(parallel loop present(insrc_indices [0:ntar],
+                                         tar_data [0:ndata],
+                                         insrc_buf_ [0:n_insrc_buf]) if (_nt->compute_gpu)
+                       async(_nt->stream_id))
+    nrn_pragma_omp(target teams distribute parallel for simd map(to: tar_indices[0:ntar]) if(_nt->compute_gpu))
     for (size_t i = 0; i < ntar; ++i) {
         tar_data[tar_indices[i]] = insrc_buf_[insrc_indices[i]];
     }
 }
 
+/// TODO: Corresponding exit data cluase for OpenACC/OpenMP is missing and hence
+///       GPU buffers are not freed.
```

Comment:
Address or create follow-up issue.

Reply:
comment: follow-up is fine, this is true for OpenACC implementation as well

Reply:
https://github.com/BlueBrain/CoreNeuron/issues/727